juju/juju
# [2.8] Ensure linter errors are always reported

## Description of change

Due to a small bug, the output of the linter steps was being suppressed. As a result, some of the code that has already landed had linter issues.

This PR fixes the problem and addresses any detected linter warnings. In addition, the `ineffassignment` linter has been enabled for test code as tests might not work as expected when the returned errors are not properly checked.

## QA steps

See the linter output on GH and/or run this locally:

```sh
$  STATIC_ANALYSIS_JOB=test_static_analysis_go make static-analysis
```
